Dependencies/UrlLib/Source/Android/UrlRequest.cpp

```diff
@@ -0,0 +1,156 @@
+                        case UrlResponseType::String:
+                        {
+                            LoadAsset(assetsManager, path.data(), m_responseString);
+                            break;
+                        }
+                        case UrlResponseType::Buffer:
+                        {
+                            LoadAsset(assetsManager, path.data(), m_responseBuffer);
+                            break;
+                        }
+                    }
+                }
+                else
+                {
+                    URL url{m_url.data()};
+
+                    URLConnection connection{url.OpenConnection()};
+                    connection.Connect();
+
+                    m_statusCode = static_cast<UrlStatusCode>(((HttpURLConnection)connection).GetResponseCode());
+
+                    InputStream inputStream{connection.GetInputStream()};
+                    ByteArrayOutputStream byteArrayOutputStream{};
+
+                    ByteArray byteArray{4096};
+                    int bytesRead{};
+                    while ((bytesRead = inputStream.Read(byteArray)) != -1)
+                    {
+                        byteArrayOutputStream.Write(byteArray, 0, bytesRead);
+                    }
+
+                    switch (m_responseType)
+                    {
+                        case UrlResponseType::String:
+                        {
+                            // TODO: use the charset from the content type?
+                            m_responseString = byteArrayOutputStream.ToString("UTF-8");
+                            break;
+                        }
+                        case UrlResponseType::Buffer:
+                        {
+                            m_responseBuffer = byteArrayOutputStream.ToByteArray();
+                            break;
+                        }
+                    }
+
+                    // Must happen after getting the content to get the redirected URL.
+                    m_responseUrl = connection.GetURL().ToString();
+                }
+            });
+        }
+
+        UrlStatusCode StatusCode() const
+        {
+            return m_statusCode;
+        }
+
+        gsl::cstring_span<> ResponseUrl()
+        {
+            return m_responseUrl;
+        }
+
+        gsl::cstring_span<> ResponseString()
+        {
+            return m_responseString;
+        }
+
+        gsl::span<const std::byte> ResponseBuffer() const
+        {
+            return m_responseBuffer;
+        }
+
+    private:
+        arcana::cancellation_source m_cancellationSource{};
+        UrlResponseType m_responseType{UrlResponseType::String};
+        UrlMethod m_method{UrlMethod::Get};
+        std::string m_url{};
+        UrlStatusCode m_statusCode{UrlStatusCode::None};
+        std::string m_responseUrl{};
+        std::string m_responseString{};
+        std::vector<std::byte> m_responseBuffer{};
+    };
+}
+
+#include <Shared/UrlRequest.h>
```

Comment:
Weird, but clever. 👍 😃

Reply:
I don't about the clever part, but weird, yes. 😄